### DISCUSSION TITLE
codegeneration using MPC

### DISCUSSION THREAD

**Discussion by ChandanGirish** [OpenDDS/OpenDDS#4229] [Q&A]
Hi Team,

I was going through getting started guide and it says codegenration using MPC, I have few questions:

- How do I create my own example_code (similar to one in OpenDDS-master/DevGuideExamples/DCPS/Messeanger) with new idl file and then build and run it? what is the procedure? can I just create .mpc in my folder and then build?

**Comment by jrw972** [comment]
I assume you are talking about [this](https://opendds.readthedocs.io/en/latest-release/devguide/getting_started.html) getting started guide.

You can copy any example in the repository and modify it to meet your needs.  Regarding IDL, you can see where `Messenger.idl` appears in `Messenger.mpc`.  If you were to rename the IDL file, you would need to update the MPC file and some includes to suit.

For building, you need to invoke `mwc.pl`.  On my Mac, that looks like `source setenv.sh;  cd DevGuideExamples/DCPS/Messenger; mwc.pl -type gnuace`.  It may be the same or different depending on your platform.

If there are particular things that the guide doesn't cover, please consider opening a pull request to add the necessary clarifications.
